packages/dds/tree/src/schema-stored/Stored and View Schema.md

```diff
@@ -35,104 +34,116 @@ Places we can store stored schema information:
 
-While a schema must be added to the document as a stored schema to use the type
+While a schema must be added to the document as a stored schema to use the type TODO: Which type? I.e. does this mean the same thing as 'a schema must be added to the document as a stored schema in order to be applied to a specific type of node'?
 (otherwise adding a new type could break existing data which might not even be downloaded on the current client),
 it's possible to add a schema that is compatible with all possible data (assuming the children themselves are compatible with their own types).
+
 Applications which wish to rely entirely on schema-on-read for some or all of their data can use this pattern for all `stored schema` and only use their actual developer authored schema as `view schema`:
 
 ## Use as `Stored Schema`
 
--   Document load: The application can check that their view schema match stored schema, and check that the root has supported content.
+TODO: Use what as stored schema? Isn't stored schema just that or can it be used as something else? Or is 'Usages of Stored Schema` a more accurate title?
+
+-   Document load: The application can check that their view schema matches their stored schema and that the root has supported content.
+
     See `checkCompatibility` in [Schema.ts](./Schema.ts) for an example of how this could work.
--   Writers: When inserting new content, and updating modified nodes, share-tree can check them against the schema in the document.
-    If the schema is missing, it must be added as part of the edit: for this the editing API needs to take in types which can have their desired stored schema queried.
--   Change application can access schema information: changes can conflict if they violate schema (or maybe only check at end of transaction? Or at special marked places and at end?).
-    Change application could (someday) adjust behavior based on schema (ex: provide set semantics to a sequence) or have schema specific edits,
-    but initial version will just detect violations and mark as conflicted.
--   We can ways to update existing schema:
-    -   Could support OP that changes a schema in a way where all data that was accepted by the old schema is accepted by the new one (ex: add optional fields (if compatible with extra fields), move required field into extraFields, add a type to a field).
-    -   Could even do things like apply a change to all nodes with a specific type to enable edits which modify a schema and update data atomically (ex: alter table like from sql). This does not work with partial checkouts well.
+-   Writers: When inserting new content and updating modified nodes, shared-tree can check them against the schema in the document.
+
+    If the schema is missing, it must be added as part of the edit: for this, the editing API needs to take in types which can have their desired stored schema queried. TODO: This was a bit confusing for me, do you have an example?
```

Comment:
Question

Reply:
If you insert something into a document that does not currently have schema for the types you are using, you also have to include the schema.

For example, if you add support for tables to whiteboards, old documents will not have this schema, so part of inserting a table will require updating the schema to allow them (it the document is explicit about which item types are allowed), as well as adding the table schema itself.

The portion of this sentence after ":" I think is noting how the editing APIs will need to be able to look up stored schema to do this. I think that's probably better to just remove since it's not clear what level of abstraction the APIs in question are at, and the reader could be thinking on one where schema updates are an implementation detail.